superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -316,6 +316,15 @@ function dbReducer(
 
 const DEFAULT_TAB_KEY = '1';
 
+const serializeExtra = (extraJson: DatabaseObject['extra_json']) =>
+  JSON.stringify({
+    ...extraJson,
+    metadata_params: JSON.parse((extraJson?.metadata_params as string) || '{}'),
+    engine_params: JSON.parse((extraJson?.engine_params as string) || '{}'),
+    schemas_allowed_for_csv_upload:
+      (extraJson?.schemas_allowed_for_csv_upload as string) || '[]',
+  });
```

Comment:
is this something that we can handle in the db reducer so that we can keep the state in the reducer as close to the state that the api requires?

Reply:
we tried this intially but would get error everytime the user would have a misformatted JSON. So we broke it out to seperate field then onCreate or testConn we'd serialize and create the request.